spark/src/main/scala/org/apache/spark/sql/comet/CometNativeScanExec.scala

```diff
@@ -60,415 +50,34 @@ case class CometNativeScanExec(
-    driverMetrics("metadataTime") = timeTakenMs
-    ret
-  }.toArray
-
-  // We can only determine the actual partitions at runtime when a dynamic partition filter is
-  // present. This is because such a filter relies on information that is only available at run
-  // time (for instance the keys used in the other side of a join).
-  @transient private lazy val dynamicallySelectedPartitions: Array[PartitionDirectory] = {
-    val dynamicPartitionFilters = partitionFilters.filter(isDynamicPruningFilter)
-
-    if (dynamicPartitionFilters.nonEmpty) {
-      val startTime = System.nanoTime()
-      // call the file index for the files matching all filters except dynamic partition filters
-      val predicate = dynamicPartitionFilters.reduce(And)
-      val partitionColumns = relation.partitionSchema
-      val boundPredicate = Predicate.create(
-        predicate.transform { case a: AttributeReference =>
-          val index = partitionColumns.indexWhere(a.name == _.name)
-          BoundReference(index, partitionColumns(index).dataType, nullable = true)
-        },
-        Nil)
-      val ret = selectedPartitions.filter(p => boundPredicate.eval(p.values))
-      setFilesNumAndSizeMetric(ret, false)
-      val timeTakenMs = (System.nanoTime() - startTime) / 1000 / 1000
-      driverMetrics("pruningTime") = timeTakenMs
-      ret
-    } else {
-      selectedPartitions
-    }
-  }
-
-  // exposed for testing
-  lazy val bucketedScan: Boolean = originalPlan.bucketedScan
-
-  override lazy val (outputPartitioning, outputOrdering): (Partitioning, Seq[SortOrder]) =
-    (originalPlan.outputPartitioning, originalPlan.outputOrdering)
-
-  @transient
-  private lazy val pushedDownFilters = getPushedDownFilters(relation, dataFilters)
-
-  override lazy val metadata: Map[String, String] =
-    if (originalPlan == null) Map.empty else originalPlan.metadata
-
-  override def verboseStringWithOperatorId(): String = {
-    val metadataStr = metadata.toSeq.sorted
-      .filterNot {
-        case (_, value) if (value.isEmpty || value.equals("[]")) => true
-        case (key, _) if (key.equals("DataFilters") || key.equals("Format")) => true
-        case (_, _) => false
-      }
-      .map {
-        case (key, _) if (key.equals("Location")) =>
-          val location = relation.location
-          val numPaths = location.rootPaths.length
-          val abbreviatedLocation = if (numPaths <= 1) {
-            location.rootPaths.mkString("[", ", ", "]")
-          } else {
-            "[" + location.rootPaths.head + s", ... ${numPaths - 1} entries]"
-          }
-          s"$key: ${location.getClass.getSimpleName} ${redact(abbreviatedLocation)}"
-        case (key, value) => s"$key: ${redact(value)}"
-      }
-
-    s"""
-       |$formattedNodeName
-       |${ExplainUtils.generateFieldString("Output", output)}
-       |${metadataStr.mkString("\n")}
-       |""".stripMargin
-  }
-
-  lazy val inputRDD: RDD[InternalRow] = {
```

Comment:
Shouldn't we return an rdd here (perhaps CometExecRDD) because in CometNativeExec we will call executeColumnar which will use this?

Reply:
No, we won't call executeColumnar  on CometNativeScanExec anymore. It is similar to other native operators.